tests/HttpCache/SouinPurgerTest.php

```diff
@@ -0,0 +1,293 @@
+            }
+
+            public function request($method, $uri, array $options = []): ResponseInterface
+            {
+                $this->requests[] = [$method, $uri, $options];
+
+                return new Response();
+            }
+
+            public function requestAsync($method, $uri, array $options = []): PromiseInterface
+            {
+                throw new LogicException('Not implemented');
+            }
+
+            public function getConfig($option = null)
+            {
+                throw new LogicException('Not implemented');
+            }
+        };
+        $loggerProphecy = $this->prophesize(LoggerInterface::class);
+
+        $purger = new SouinPurger([$client1, $client2], $loggerProphecy->reveal(), 'http://dummy_host', '/dummy_api_path/souin_api', 'souin_auth_cookie');
+        $purger->purge(['/foo']);
+        self::assertSame([
+            'PURGE',
+            'http://dummy_host/dummy_api_path/souin_api',
+            [
+                'headers' => [
+                    'Surrogate-Key' => '/foo',
+                    'Cookie' => 'souin-authorization-token=souin_auth_cookie',
+                ],
+            ],
+        ], $client1->requests[0]);
+        self::assertSame([
+            'PURGE',
+            'http://dummy_host/dummy_api_path/souin_api',
+            [
+                'headers' => [
+                    'Surrogate-Key' => '/foo',
+                    'Cookie' => 'souin-authorization-token=souin_auth_cookie',
+                ],
+            ],
+        ], $client2->requests[0]);
+    }
+
+    public function testPurgeWithErrorOnClient()
+    {
+        $guzzleExceptionProphecy = $this->prophesize(ClientException::class);
+        $client = new class($guzzleExceptionProphecy->reveal()) implements ClientInterface {
+            public $request = [];
+            private $guzzleExceptionProphecy;
+
+            public function __construct($guzzleExceptionProphecy)
+            {
+                $this->guzzleExceptionProphecy = $guzzleExceptionProphecy;
+            }
+
+            public function send(RequestInterface $request, array $options = []): ResponseInterface
+            {
+                throw new LogicException('Not implemented');
+            }
+
+            public function sendAsync(RequestInterface $request, array $options = []): PromiseInterface
+            {
+                throw new LogicException('Not implemented');
+            }
+
+            public function request($method, $uri, array $options = []): ResponseInterface
+            {
+                throw $this->guzzleExceptionProphecy;
+            }
+
+            public function requestAsync($method, $uri, array $options = []): PromiseInterface
+            {
+                throw new LogicException('Not implemented');
+            }
+
+            public function getConfig($option = null)
+            {
+                throw new LogicException('Not implemented');
+            }
+        };
```

Comment:
I don't really get why the whole interface needs to be implemented

Reply:
Because I need a ClientInterface.